package/network/services/openvpn/files/lib/netifd/openvpn-up

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+#DBG=1
+
+[ -n "$INTERFACE" ] || {
+	echo "openvpn-up no INTERFACE variable appended"
+	exit 1
+}
+
+[ -n "$DBG" ] && {
+	env > /tmp/openvpn-up-${INTERFACE}-env.log
+}
+
+. /lib/netifd/netifd-proto.sh
+proto_init_update "$dev" 1 1
+proto_set_keep 1
+[ -n "$ifconfig_local" ] && proto_add_ipv4_address "$ifconfig_local" 32
+[ -n "$ifconfig_ipv6_local" ] && proto_add_ipv6_address "$ifconfig_ipv6_local" 128
+
+# set ipv4 routen
+c=1
+while true ; do
+	v=route_network_$c
+	eval "network=\$$v"
+	[ -n "$network" ] || break
+	v=route_gateway_$c
+	eval "gw=\$$v"
+	v=route_netmask_$c
+	eval "mask=\$$v"
+	v=route_metric_$c
+	eval "metric=\$$v"
+	[ -n "$metric" ] || metric="$METRIC"
+
+	proto_add_ipv4_route "$network" "$mask" "$gw" "" "$metric"
+	c=$(( $c+1 ))
+done
+
+# set default route
+[ -n "$DEFAULTROUTE" ] && {
+	eval "metric=\$$METRIC"
+	proto_add_ipv4_route "0.0.0.0" "0" "$gw" "0.0.0.0/0" "$metric"
```

Comment:
@jow i am not sure if this works for all situations the $gw variable is in my setup always the same.